modules/javafx.graphics/src/main/java/com/sun/javafx/css/BitSet.java

```diff
@@ -539,31 +543,31 @@ public boolean equals(Object obj) {
     private boolean equalsBitSet(BitSet<?> other) {
         int a = this.bits != null ? this.bits.length : 0;
         int b = other.bits != null ? other.bits.length : 0;
+        int max = Math.max(a, b);
 
-        if (a != b) return false;
+        for (int i = 0; i < max; i++) {
+            long m0 = i >= a ? 0 : this.bits[i];
+            long m1 = i >= b ? 0 : other.bits[i];
 
-        for(int m=0; m<a; m++) {
-            final long m0 = this.bits[m];
-            final long m1 = other.bits[m];
             if (m0 != m1) {
                 return false;
             }
         }
+
         return true;
     }
 
-    abstract protected T getT(int index);
-    abstract protected int getIndex(T t);
+    protected abstract T getT(int index);
+    protected abstract int getIndex(T t);
 
-    /*
-     * Try to cast the arg to a T.
-     * @throws ClassCastException if the class of the argument is
-     *         is not a T
-     * @throws NullPointerException if the argument is null
+    /**
+     * Returns the element type.
+     *
+     * @return a {@link Class} of type {@code T}, never {@code null}
      */
-    abstract protected T cast(Object o);
+    protected abstract Class<T> getElementType();
 
-    protected long[] getBits() {
+    long[] getBits() {
```

Comment:
Is there any reason for this change (given that the abstract methods are still protected)?

Reply:
The subclasses no longer need this, and so I reduced its scope.  I would reduce it further, but tests are using it.